README-relationships.md

```diff
@@ -0,0 +1,262 @@
+# Modeling relationships in the introspection schema
+
+## Background
+
+The Model Introspection Schema (MIS) is an intermediate representation of the GraphQL model that includes Amplify annotations. It is different
+from the standard [GraphQL introspection schema](https://graphql.org/learn/introspection/) in that it includes relationship information, not
+just type information.
+
+> **NOTE:** The MIS is an internal implementation detail of the Amplify API plugin. It should not be used in a customer application.
+
+## Sample
+
+Given a schema like
+
+```graphql
+type Primary @model @auth(rules: [{ allow: public, operations: [read] }, { allow: owner }]) {
+  id: ID! @primaryKey
+  relatedMany: [RelatedMany] @hasMany(references: "primaryId")
```

Comment:
Is non-array input for references supported? It is currently not supported for `fields`.

Reply:
Yes, it is automatically cast to an array.